hypershift-operator/controllers/hostedcluster/network_policies.go

```diff
@@ -161,6 +168,15 @@ func (r *HostedClusterReconciler) reconcileNetworkPolicies(ctx context.Context,
 				}); err != nil {
 					return fmt.Errorf("failed to reconcile konnectivity nodeport network policy: %w", err)
 				}
+
+				// Reconcile nodeport-konnectivity Network Policy when konnectivity is hosted in the kas pod
+				policy = networkpolicy.NodePortKonnectivityKASNetworkPolicy(controlPlaneNamespaceName)
+				if _, err := createOrUpdate(ctx, r.Client, policy, func() error {
```

Comment:
I think this will create the policy always, we need to identify where is konnectivity and then place the netpol accordingly.

Reply:
> we need to identify where is konnectivity and then place the netpol accordingly.

how? the hypershift operator isn't responsible for laying down the konnectivity agent.